### DISCUSSION TITLE
EfficientAd detects defects on the masked areas

### DISCUSSION THREAD

**Discussion by crazyblueer** [open-edge-platform/anomalib#2217] [Q&A]
I am using EfficientAd and I see that when inferencing, it tends to focus on objects with color within the image and not focus on the black region - where I need it to do the detection task. Since those colored objects are unnecessary, I masked them with white color using OpenCV, and then use Fuzzy contrast and CLAHE contrast to highlight the defects (since my defects are quite weak). However, when I performed inference again, the model highlighted the left side of those masked regions as defects. I don't really understand why since I assumed that if the masking is applied to all images, the model should ignore those masked regions. I am not sure if I haven't done the preprocessing step right or maybe because of some properties with white color. I tried to play around with the order of equalization histograms and masking, but both yields the same results. 
 
I'd appreciate any insights on this. Thank you!

**Comment by abc-125** [comment]
Hello, it is hard to tell without seeing images. If this masked area changes size / shape / location, it can be seen as anomalous by EfficientAD. I would focus at one problem at a time and drop all image transformations (Fuzzy contrast and CLAHE) and try to see if this changes anything, also try to use gray color instead of white.

**Reply by crazyblueer** [reply]
Thank you very much! maybe the masking did affect the sizes and the locations of the object. I decided to filter this defect out when using Detection Mode because I can't find another way to make the masking equal in size in every image since the object might be rotated during production. However, without image enhancement, I am afraid the model can't "see" the defects because they are so weak and almost resemble the background. Do you have any recommendation on how to fine tune or maybe some other models that work better with weak defects?

**Reply by abc-125** [reply]
You're welcome! 

> I am afraid the model can't "see" the defects because they are so weak and almost resemble the background

Have you tried this without any filters? Most of anomaly detection models rely on feature extractors, pretrained on ImageNet (regular images without strong filters), so I rather would expect strong filters decrease the performance.
